extensions/hawkbit-extension-artifact-repository-mongo/src/main/java/org/eclipse/hawkbit/artifact/repository/MongoDBArtifactStore.java

```diff
@@ -143,18 +143,20 @@ private void deleteArtifact(final GridFSDBFile dbFile) {
 
     }
 
-    private DbArtifact store(final InputStream content, final String contentType, final OutputStream os,
-            final File tempFile, final DbArtifactHash hash) {
+    private DbArtifact store(final String tenant, final InputStream content, final String contentType,
+            final OutputStream os, final File tempFile, final DbArtifactHash hash) {
         final GridFsArtifact storedArtifact;
         try {
             final String sha1Hash = computeSHA1Hash(content, os, hash != null ? hash.getSha1() : null);
             // upload if it does not exist already, check if file exists, not
             // tenant specific.
-            final GridFSDBFile result = gridFs.findOne(new Query().addCriteria(Criteria.where(FILENAME).is(sha1Hash)));
+            final GridFSDBFile result = gridFs.findOne(new Query()
+                    .addCriteria(Criteria.where(FILENAME).is(sha1Hash).and(TENANT_QUERY).is(tenant.toUpperCase())));
```

Comment:
In this method the `tenant.toUpperCase()` is called twice. You could make to it upper-case one time and store in a local variable. Because the tenant string needs to be sanitized across the class it's maybe nicer to introduce a private method 
```
public String sanitizeTenant(final String tenant)
{
   return tenant.toUpperCase();
}
```